azurerm/internal/services/web/app_service.go

```diff
@@ -1722,6 +1868,40 @@ func flattenAppServiceSiteConfig(input *web.SiteConfig) []interface{} {
 	return append(results, result)
 }
 
+func flattenAutoHealRules(input *web.AutoHealRules) interface{} {
+	rules := make([]interface{}, 0)
+	if input == nil {
+		return rules
+	}
+
+	rule := make(map[string]interface{})
+
+	triggers := input.Triggers
+	triggerVals := make(map[string]interface{})
+	if requests := triggers.Requests; requests != nil {
+		vals := make(map[string]interface{})
+		vals["count"] = requests.Count
+		vals["time_interval"] = requests.TimeInterval
+		triggerVals["requests"] = vals
+	}
+
+	if slowRequests := triggers.SlowRequests; slowRequests != nil {
+		vals := make(map[string]interface{})
+		vals["count"] = slowRequests.Count
+		vals["time_interval"] = slowRequests.TimeInterval
+		vals["time_taken"] = slowRequests.TimeTaken
+		triggerVals["slow_requests"] = vals
+	}
+
+	actions := input.Actions
+	actionVals := make(map[string]interface{})
+	actionVals["action_type"] = actions.ActionType
+	actionVals["min_process_execution_time"] = actions.MinProcessExecutionTime
+	rule["actions"] = actionVals
```

Comment:
This will need to be reworked a little to check values are not nil before assignment (e.g. input.Triggers is a pointer, that may be nil). We also need to ensure that every expected property has a value set.  You should be able to use `flattenAppServiceLogs` as an example of flattening this type of structure.
We also appear to be missing the status codes section here?